sdk/cognitiveservices/azopenai/policy_apikey.go

```diff
@@ -14,8 +14,19 @@ import (
 
 // KeyCredential is used when doing APIKey-based authentication.
 type KeyCredential struct {
-	// APIKey is the api key for the client.
-	APIKey string
+	// apiKey is the api key for the client.
+	apiKey string
+}
+
+// KeyCredentialOptions contains the optional parameters for the [NewKeyCredential] method.
+type KeyCredentialOptions struct {
+	// For future expansion
+}
+
+// NewKeyCredential creates a KeyCredential containing an API key for
+// either Azure OpenAI or OpenAI.
+func NewKeyCredential(apiKey string, options *KeyCredentialOptions) (KeyCredential, error) {
```

Comment:
I don't think we offer options params for credential types. @chlowell should review this.

Reply:
Just spot checking, but it looks like we do:

```go
func NewAzureCLICredential(options *AzureCLICredentialOptions) (*AzureCLICredential, error)
func NewDefaultAzureCredential(options *DefaultAzureCredentialOptions) (*DefaultAzureCredential, error)
func NewDeviceCodeCredential(options *DeviceCodeCredentialOptions) (*DeviceCodeCredential, error)

```

Reply:
For `azblob` we don't https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/storage/azblob/common.go#L19

Reply:
TBH, with this library, I'd want to make it as "future flexible" as I can. Feels like adding it is minimal fuss (like any of our `options` taking functions) and gives us some insulation.

Reply:
Richard and I were having a side conversation about this, I'll summarize.

The credential types in `azidentity` are higher-level/more complex, covering a wide range of scenarios.  Thus, the need for options types makes sense.

For service-specific key credential types, I would think that we know the shape of the credential which would obviate the need for any options.

Given the rapid change in the OpenAI endpoint, how likely is it that the shape of the key credential will change?

Reply:
For now I'll just remove it, but we should keep it in mind. 

This also is just a beta, we have time to consider it.